app/views/layouts/application.html.erb

```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html lang="<%= LocalesService.current %>">
```

Comment:
The reason why I refrained from adding this `lang` attribute originally is because the site will sometimes operate in a multi-lingual (multi-languages?) mode – all site content is only in English, but course content can be switched to a different language. If we are to add this attribute, should it be the site content language or the course content language?

Maybe there's a better pattern for supporting a multi/dual language site?

Reply:
Done some research...

So we need this lang attribute, because thats how screenreaders know how to pronounce the content on the page.
Currently if a user changes their language on the course page, this should update. However it doesn't because turbolinks prevents the html tag from being rerendered. It is only updated if the user actually refreshes the page.

I've raised an issue with turbolinks, but don't expect a quick fix.

Playing around with the screen reader on my machine if the lang='en' and I am displayed some arabic content it won't say anything, but if lang='ar' and I'm displayed english content the screen reader will at least try!

Recommendation is that we add specific lang attributes to those elements that are translated.

Reply:
> However it doesn't because turbolinks prevents the html tag from being rerendered. It is only updated if the user actually refreshes the page.
>
> I've raised an issue with turbolinks, but don't expect a quick fix

Ah yeah, I remember that Turbolinks issue – another reason why adding the `lang` attribute wouldn't have / won't work as needed. Thanks for raising the issue with them; you may have seen https://github.com/turbolinks/turbolinks/issues/372, which indicates that they're unlikely to change this :(

One thought: we could [disable turbolinks on the language switching links](https://github.com/turbolinks/turbolinks#disabling-turbolinks-on-specific-links) to get by this problem?

> Recommendation is that we add specific lang attributes to those elements that are translated.

Great!

Reply:
Think what I've done is better than turning turbolinks off because now we've got the whole site labelled as being in english and then any parts that could be translated will have their lang attribute updated accordingly.

Reply:
Makes a lot of sense!